### PR TITLE
#63 Update SignUp Form and Enable User Profile Picture Uploading

### LINKED ISSUE #63: Update SignUp Form
#### Major Updates
- Add college, contact methods info to sign up form
- Enable user to upload profile picture when they sign up

#### Fixed Bugs
- Form was submitted even if required fields are not filled
- Request Ride not connected with uid correctly

### LINKED ISSUE #73: Bugs: Search on NavBar doesn't work
When u click on login and immediately click on search (navbar). Program breaks
Preview
recordIt

### PR DESCRIPTION
**Pull request by EvelynL09**
resolves #63
### Note
- User's profile picture is now available from a user document that you fetch from `/api/users/:userid` @micheyale @alex-dinh @JulesFL 
    - `var user = { photoURL, ...rest };`
    - You can display it with ` <Image src={photoURL} avatar/>`
    - photoURL will be set to a defualt url if the user didn't provide any picture
- `required` keyword for `<Dropdown>` component is not working, so I manually validate the school field to make sure it's not empty.
### Demo
demo_signUpForm
### Bug Fix
- Fix the issue that requestRide not connected with the correct uid.
- Fix #73

### CODE REVIEW

**Comment by micheyale** [comment]
@EvelynL09 Let's have a default url profile picture stored in DB and use a default image with an avatar on it. I think having an avatar 
 character image is represents more of a profile picture than that photoframe image semantic ui provides. 
Also, did u use Dropdown or Form.Dropdown? Someone posted the same question on stack overflow
 https://stackoverflow.com/questions/56843792/way-to-make-dropdown-form-fields-required-in-semantic-ui-react

File: frontend/src/components/Auth/SignUp.js [hunk 1]
```diff
@@ -56,72 +108,151 @@ class SignUp extends React.Component {
...
+            schoolOptions,
+            searchQuery,
+            photoURL
         } = this.state;
         return (
             <div>
                 {/* TODO: setup redirect after new user created */}
                 <Form onSubmit={this.handleSubmit}>
+                    <center>
+                    <Button as="label" htmlFor="file" type="button" size='tiny' circular>
```
Comment (micheyale) [inline frontend/src/components/Auth/SignUp.js:245]:
```suggestion
                    <Button as="label" htmlFor="file" type="button" size="tiny" circular>
```

File: frontend/src/components/Auth/SignUp.js [hunk 2]
```diff
@@ -56,72 +108,151 @@ class SignUp extends React.Component {
...
+                        <Image src={photoURL} size='tiny' circular />
```
Comment (micheyale) [inline frontend/src/components/Auth/SignUp.js:246]: same here

File: frontend/src/components/Auth/SignUp.js [hunk 3]
```diff
@@ -146,6 +277,20 @@ class SignUp extends React.Component {
                         onChange={this.handleChange}
                         required
                     />
+                    <Form.Dropdown
+                        label="College Name"
+                        options={schoolOptions}
+                        placeholder='Select School'
```
Comment (micheyale) [inline frontend/src/components/Auth/SignUp.js:283]:
```suggestion
                        placeholder="Select School"
```

File: frontend/src/components/Auth/SignUp.js [hunk 4]
```diff
@@ -1,35 +1,87 @@
...
+    handleContactMethodAddition = (e, { value }) => {
+        this.setState((prevState) => ({
+            schoolOptions: [{ text: value, value }, ...prevState.schoolOptions],
+        }))
+    };
+
+    handleSchoolAutoComplete = (e, { searchQuery }) => {
+        this.setState({ searchQuery });
+        const encodedSearchWords = encodeURI(searchQuery);
+        fetch(`/api/colleges/${encodedSearchWords}`)//TODO: try fetch the whole list all at oncce
```
[the remark this answers is not in the document]
Comment (EvelynL09) [reply frontend/src/components/Auth/SignUp.js]: I tried to fetch the wholelist all at once, but it's still slow for react to 'react' lol. So I limit the query result to be  10, and the option list is loaded way faster. Could you test it again and lemme know how to you feel about the current solution?

[the remark this answers is not in the document]
Comment (micheyale) [reply frontend/src/components/Auth/SignUp.js]: okay lemme merge this into the master and try it out

**Pull request merged**